ravendb/rvn_server.go

```diff
@@ -0,0 +1,824 @@
+	store, err := getStore(sc, 0)
+	if err != nil {
+		return "", err
+	}
+
+	err = sc.deployRavenDbInstances(parallel)
+	if err != nil {
+		return "", err
+	}
+
+	err = sc.addNodesToCluster(store)
+	if err != nil {
+		return "", err
+	}
+
+	clusterTopology, err := sc.getClusterTopology(store)
+	if err != nil {
+		return "", err
+	}
+
+	err = sc.getDatabaseHealthCheck(store)
+	if errors.As(err, &databaseDoesNotExistError) {
+		err = sc.createDb(store)
+		if err != nil {
+			return "", err
+		}
+	} else if err != nil {
+		return "", err
+	}
+
+	return clusterTopology.Topology.TopologyID, nil
+}
+
+func (sc *ServerConfig) getDatabaseHealthCheck(store *ravendb.DocumentStore) error {
+	databaseHealthCheck := operations.OperationDatabaseHealthCheck{}
+	err := executeWithRetriesMaintenanceOperations(store, &databaseHealthCheck)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sc *ServerConfig) getClusterTopology(store *ravendb.DocumentStore) (operations.OperationGetClusterTopology, error) {
+	clusterTopology := operations.OperationGetClusterTopology{}
+	err := executeWithRetries(store, &clusterTopology)
+	if err != nil {
+		return operations.OperationGetClusterTopology{}, err
+	}
+	return clusterTopology, nil
+}
+
+func getStore(config *ServerConfig, index int) (*ravendb.DocumentStore, error) {
+	var host string
+
+	host = config.Url.List[index]
+
+	serverNode := []string{host}
+	store := ravendb.NewDocumentStore(serverNode, config.HealthcheckDatabase)
+
+	if config.Insecure == false {
+		key, crt, err := utils.PfxToPem(config.ClusterCertificate)
+
+		if err != nil {
+			return nil, err
+		}
+
+		cert, err := tls.X509KeyPair(crt, key)
+		if err != nil {
+			return nil, err
+		}
+
+		x509cert, err := x509.ParseCertificate(cert.Certificate[0])
+		if err != nil {
+			return nil, err
+		}
+		store.TrustStore = x509cert
+		store.Certificate = &cert
+	}
+
+	if err := store.Initialize(); err != nil {
+		return nil, err
+	}
+
+	return store, nil
+}
+
+func (sc *ServerConfig) addNodesToCluster(store *ravendb.DocumentStore) error {
+	clusterTopology, err := sc.getClusterTopology(store)
+	var errAllDown *ravendb.AllTopologyNodesDownError
+	if errors.As(err, &errAllDown) {
+		for i := 1; i < len(sc.Url.List); i++ {
+			err = addNodeToCluster(store, sc.Url.List[i])
+			if err != nil {
+				return err
+			}
+		}
+	} else if err != nil {
+		return err
+	}
+
+	for _, nodeUrl := range clusterTopology.Topology.AllNodes {
+		var nodTag string
+		if contains(sc.Url.List, nodeUrl) {
+			continue
+		} else {
+			parse, err := url.Parse(nodeUrl)
+			if err != nil {
+				return err
+			}
+			if sc.Insecure == false {
+				hostName := strings.Split(parse.Host, ".")
+				tag := strings.ToUpper(hostName[0])
+				err = executeWithRetries(store, &operations.RemoveClusterNode{
+					Node: nodeUrl,
+					Tag:  tag,
+				})
+				if err != nil {
+					return err
+				}
+			} else {
+				for key, val := range clusterTopology.Topology.Members {
```

Comment:
Shouldn't you try to get the `tag` from the host here as well?
Why does it matter if this is secured or not?